tests/stdlib/thttpclient.nim

```diff
@@ -112,6 +114,40 @@ proc syncTest() =
   except:
     doAssert false, "TimeoutError should have been raised."
 
-syncTest()
+proc makeIPv6HttpServer(hostname: string, port: Port): AsyncFD =
+  let fd = newNativeSocket(AF_INET6)
+  setSockOptInt(fd, SOL_SOCKET, SO_REUSEADDR, 1)
+  var aiList = getAddrInfo(hostname, port, AF_INET6)
```

Comment:
Yeah, I thought about it, but for servers the situation is slightly more complicated. The only way (that I am aware about) to listen to both IPv4 and IPv6 connections on single socket is to bind to in6addr_any constant, and set IPV6_V6ONLY socket flag to zero. However, binding to in6addr_any means you would be listening on every available interface, which is not good if you would like e.g. to only listen locally or on a specific interface. This setup would force you to configure firewalls most of the time.

Another option is two have two sockets (one for IPv4 and IPv6) but then we would have to somehow abstract it away so that it looks like one socket for the API client.

I was not aware about Go listen. Quick googling reveals that it simply binds to the first available address that getAddrInfo returns and can be bound to (pretty much like our dial does with connect). If we are happy with that, then those "setupServerSocket" procs from the tests can be easily translated into `listen`. But let's do that separately from this PR.